validation/validation.go

```diff
@@ -0,0 +1,370 @@
+package validation
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+	"regexp"
+	"slices"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/gofrs/uuid/v5"
+)
+
+// New is the main function in charge of validating the HTTP request form by using the defined rule set.
+// Form fields will be validated only if there is a rule that indicates they must be validated.
+func New(form url.Values, ruleSet ...Rule) map[string][]string {
+	verrs := make(map[string][]string)
+	mutex := new(sync.RWMutex)
+
+	for _, rule := range ruleSet {
+		mutex.Lock()
+		verrs[rule.Field] = append(verrs[rule.Field], rule.validate(form[rule.Field]...)...)
+		mutex.Unlock()
+	}
+
+	for k, v := range verrs {
+		if len(v) == 0 {
+			delete(verrs, k)
+		}
+	}
+
+	return verrs
+}
+
+// Validation is a condition that must be satisfied by all values in a specific form field.
+// or else an error message is displayed indicating that at least one value is invalid.
+type Validation func(...string) error
+
+// Required function validates the form field has no-empty values.
+func Required(message ...string) Validation {
+	return func(values ...string) error {
+		hasEmptyValues := slices.ContainsFunc(values, func(val string) bool {
+			return strings.TrimSpace(val) == ""
+		})
+
+		if len(values) > 0 && !hasEmptyValues {
+			return nil
+		}
+
+		return newError("This field is required.", message...)
+	}
+}
+
+// Match function validates the form field values with a string.
+func Match(value string, message ...string) Validation {
```

Comment:
we could name this one `Matches` and name the first variable `field` so its clear we're matching against a field.